flink-connectors/flink-connector-hive/src/test/java/org/apache/flink/connectors/hive/HiveDialectITCase.java

```diff
@@ -918,6 +918,72 @@ public void testMacro() throws Exception {
                         badMacroName);
     }
 
+    @Test
+    public void testSetCommand() throws Exception {
+        // test set system:
+        tableEnv.executeSql("set system:xxx=5");
+        assertThat(System.getProperty("xxx")).isEqualTo("5");
+
+        // test set hiveconf:
+        tableEnv.executeSql("set hiveconf:yyy=${system:xxx}");
+        assertThat(hiveCatalog.getHiveConf().get("yyy")).isEqualTo("5");
+
+        // test set hivevar:
+        tableEnv.executeSql("set hivevar:a=1");
+        tableEnv.executeSql("set hiveconf:zzz=${hivevar:a}");
+        assertThat(hiveCatalog.getHiveConf().get("zzz")).isEqualTo("1");
+
+        // test the hivevar still exists when we renew the sql parser
+        tableEnv.getConfig().setSqlDialect(SqlDialect.DEFAULT);
+        tableEnv.executeSql("show tables");
+        tableEnv.getConfig().setSqlDialect(SqlDialect.HIVE);
+        tableEnv.executeSql("set hiveconf:zzz1=${hivevar:a}");
```

Comment:
1. Use `set` statement to switch dialect? 
2. Set a flink config under hive dialect and check it works.

Reply:
1: We maynot use `set` statement to switch dialect directly for the `SetOperation` is executed in SqlClient.
2: The Flink's `SetOperation` will be executed in SqlClient , so we can't set it in tableEnv. But the test [set.q](https://github.com/apache/flink/blob/master/flink-table/flink-sql-client/src/test/resources/sql/set.q) in flink-sql-client has covered the case that set flink config in HiveDialect.